cmake/modules/SwiftSource.cmake

```diff
@@ -441,12 +440,29 @@ function(_compile_swift_files
 
   # Then we can compile both the object files and the swiftmodule files
   # in parallel in this target for the object file, and ...
+
+  # Windows doesn't support long command line paths, of 8191 chars or over.
+  # We need to work around this by avoiding long command line arguments. This can be
+  # achieved by writing the list of file paths to a file, passing that file to a new
+  # Python utility script (line-directive-file), which reads the list of file paths,
+  # passing them to the old Python utility script, line-directive.
+  set(line_directive_tool "${SWIFT_SOURCE_DIR}/utils/line-directive-file")
+  set(line_directive_arguments, "${CMAKE_CURRENT_BINARY_DIR}/line-directive-file.txt")
+
+  set(source_files_string "")
+  foreach(file ${source_files})
+    set(source_files_string "${source_files_string} ${file}")
+  endforeach()
```

Comment:
I've moved it into `SwiftList.cmake`. I think we can add uses of the new `list_join` function as we find them